### PR TITLE
Upgrade hadoop version to 3.2.4

### PR DESCRIPTION
**Pull request by abhioncbr**
As per the [issue](https://github.com/apache/pinot/issues/11364), bumping the hadoop version to `3.2.4`

labels
- dependency

### CODE REVIEW

**Comment by abhioncbr** [comment]
@Jackie-Jiang please review. Thanks

File: pinot-plugins/pinot-input-format/pinot-orc/pom.xml [hunk 1]
```diff
@@ -46,6 +46,16 @@
       <groupId>org.apache.hadoop</groupId>
       <artifactId>hadoop-common</artifactId>
       <scope>${hadoop.dependencies.scope}</scope>
+      <exclusions>
```
Comment (Jackie-Jiang) [inline pinot-plugins/pinot-input-format/pinot-orc/pom.xml:49]: Should we add these exclusions into the root pom?

File: pinot-plugins/pinot-input-format/pinot-orc/pom.xml [hunk 2]
```diff
@@ -58,10 +68,20 @@
...
+    </dependency>
     <dependency>
       <groupId>xml-apis</groupId>
       <artifactId>xml-apis</artifactId>
       <scope>compile</scope>
     </dependency>
+    <dependency>
+      <groupId>org.codehaus.woodstox</groupId>
+      <artifactId>stax2-api</artifactId>
+      <version>4.1</version>
```
Comment (Jackie-Jiang) [inline pinot-plugins/pinot-input-format/pinot-orc/pom.xml:84]: Should we add this into the root pom?

**Review by Jackie-Jiang — COMMENTED** [review]
Thanks for taking this task. Maybe I was wrong, but IMO we should pull as much dependency management as possible into the root pom, and not overriding them in the sub module.

@jackjlli Can you also take a look at this?
cc @mcvsubbu @siddharthteotia

File: pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-hadoop/pom.xml [hunk 3]
```diff
@@ -35,7 +35,7 @@
   <url>https://pinot.apache.org/</url>
   <properties>
     <pinot.root>${basedir}/../../..</pinot.root>
-    <hadoop.version>2.10.1</hadoop.version>
+    <hadoop.version>3.2.4</hadoop.version>
```
[the remark this answers is not in the document]
Comment (jackjlli) [reply pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-hadoop/pom.xml:38]: +1

File: pinot-plugins/pinot-input-format/pinot-parquet/pom.xml [hunk 4]
```diff
@@ -76,7 +86,24 @@
...
+        <exclusion>
+          <groupId>org.eclipse.jetty</groupId>
+          <artifactId>jetty-util</artifactId>
+        </exclusion>
       </exclusions>
     </dependency>
+    <dependency>
+      <groupId>org.codehaus.woodstox</groupId>
+      <artifactId>stax2-api</artifactId>
+      <version>4.1</version>
```
Comment (jackjlli) [inline pinot-plugins/pinot-input-format/pinot-parquet/pom.xml:106]: Same here. Please add it to the root pom.

**Review by jackjlli — COMMENTED** [review]
Tested that the code change in this PR works in LinkedIn env. We should be good with this change.
Minor but LGTM.

**Comment by abhioncbr** [comment]
@Jackie-Jiang / @jackjlli, it's ready for review with the suggested changes. please review. Thanks

File: pom.xml [hunk 5]
```diff
@@ -159,6 +159,7 @@
     <jmh.version>1.26</jmh.version>
     <audienceannotations.version>0.13.0</audienceannotations.version>
     <clp-ffi.version>0.4.3</clp-ffi.version>
+    <stax2-api>4.1</stax2-api>
```
Comment (Jackie-Jiang) [inline pom.xml:162]:
```suggestion
    <stax2-api.version>4.1</stax2-api.version>
```

**Review by Jackie-Jiang — COMMENTED** [review]
Is the `pinot-plugins.tar.gz` included intentional?

**Comment by abhioncbr** [comment]
> Is the `pinot-plugins.tar.gz` included intentional?

deleted.

**Comment by abhioncbr** [comment]
Please review it once more. Thanks

**Review by Jackie-Jiang — APPROVED** [review]

**Pull request merged**